pages/index.tsx

```diff
@@ -1,30 +1,125 @@
+  | 'CodeBlock'
+  | 'ApiTable'
+
+interface HomeProps {
+  sections: any
 }
 
 export async function getStaticProps() {
-  const res = await fetchEntries()
-
-  const posts = await res.map((post: any) => {
-    return post.fields
-  })
+  const response = await fetchContent(`
+    {
+      sectionCollection(limit: 5) {
+        items {
+          title
+          articleCollection(limit: 10) {
+            items {
+              ... on Article {
+                contentCollection(limit: 10) {
+                  items {
+                    __typename
+                    ... on Hero {
+                      title
+                      heroDescription: description
+                    }
+                    ... on ContentBlock {
+                      title
+                      description {
+                        json
+                      }
+                    }
+                    ... on LiveExample {
+                      title
+                      description {
+                        json
+                      }
+                    }
+                    ... on CodeBlock {
+                      title
+                      description {
+                        json
+                      }
+                      filename
+                      sourceCode
+                    }
+                    ... on ApiTable {
+                      title
+                      apiTableDescription: description
+                      apiFieldCollection(limit: 10) {
+                        items {
+                          ... on ApiField {
+                            name
+                            dataType
+                            defaultValue
+                            description
+                            required
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `)
 
   return {
     props: {
-      posts,
+      sections: response?.sectionCollection?.items ?? [],
     },
   }
 }
+
+/**
+ * Map content types to components using adapters
+ *
+ */
+function renderPresenter(type: ContentType, fields: any): React.ReactElement {
+  const componentMap = {
+    Hero: <HeroAdapter fields={fields} />,
+    ContentBlock: <ContentBlockAdapter fields={fields} />,
+    LiveExample: <LiveExampleAdapter fields={fields} />,
+    CodeBlock: <CodeBlockAdapter fields={fields} />,
+    ApiTable: <ApiTableAdapter fields={fields} />,
+    // ...
+  }
+
+  return componentMap[type]
+}
+
+export const Home: React.FC<HomeProps> = ({ sections }) => {
+  const articleCollection = sections.flatMap(
+    (item) => item?.articleCollection?.items ?? []
+  )
+
+  const contentCollection = articleCollection.flatMap(
+    (item) => item?.contentCollection?.items ?? []
+  )
+
+  return (
+    <LayoutFramework>
+      {contentCollection.map(({ __typename, ...fields }: any) => {
```

Comment:
There are some loose `any` types in this PR because the types come from Contentful.

There is potential for us to interrogate the GraphQL API and generate types using [this](https://graphql-code-generator.com/).

We use this in some of the downstream exemplars. I've created a separate ticket: